src/dbnode/namespace/types.go

```diff
@@ -357,3 +365,21 @@ type DownsampleOptions struct {
 	// (e.g. rollup/recording rules).
 	All bool
 }
+
+// Status is the status of the namespace.
+type Status uint8
+
+const (
+	Unknown Status = iota
+	// Initializing means the namespace is in the process of coming online for use.
+	Initializing
+	// Ready means the namespace is ready for use.
+	Ready
+)
+
+// StagingState is the state associated with a namespace's
+// availability for reads and writes.
+type StagingState interface {
+	// Status returns the status of the namespace.
+	Status() Status
+}
```

Comment:
Is this just because we want to optionally set `State` on `Options`?

Reply:
That's correct. We want nil StagingState to be an option which is the expectation for namespaces created before this existed

Reply:
Could we have an additional `NamespaceStagingStatus` entry like `LegacyStagingStatus` or something along those lines, with a comment explaining it's unset in the incoming proto, to avoid creating this interface?

Reply:
How do you feel about just using `UNKNOWN` for this? Makes the handling of already existing namespaces pretty easy.